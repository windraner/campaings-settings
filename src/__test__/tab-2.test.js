import React from 'react';
import Tab2 from '../components/main-content-section/tab-block/tab-2/tab-2';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import image from '../test.png';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

const mock = {
  isMediaCollection: false,
  isPhotosModerationQueue: true,
  mediaCollectionList: ['qwe', 'rty'],
  stateHandler() {},
  switchHandler() {},
  accountsSettingsItems: [
    {
      id: '1',
      url: image,
      userName : 'Instagram',
      platform: 'Instagram',
      status: 'Verified',
      userFeed: true,
      taggedFeed: true,
      mentionedFeed: false
    },
    {
      id: '2',
      url: image,
      userName : 'Blackmirror',
      platform: 'Twitter',
      status: 'Rejected',
      userFeed: false,
      taggedFeed: false,
      mentionedFeed: false
    }
  ],
};

describe('<Tab2 />', () => {
  it('Check for media collection switcher', () => {
    const wrapper = shallow(<Tab2 {...mock} />);
    expect(wrapper.find('.campaigns-settings__border-separator').length).toBe(0);
    wrapper.setProps({ isMediaCollection: true });
    expect(wrapper.find('.campaigns-settings__border-separator').length).toBe(1);
  });
  it('matches the snapshot', () => {
    const tree = shallow(<Tab2 {...mock} />);
    expect(toJson(tree)).toMatchSnapshot();
    tree.setProps({ isMediaCollection: true });
    expect(toJson(tree)).toMatchSnapshot();
  });
});
