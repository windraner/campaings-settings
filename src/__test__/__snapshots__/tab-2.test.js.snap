// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tab2 /> matches the snapshot 1`] = `
<Fragment>
  <TabTitle
    text="Media collection"
  />
  <div
    className="campaigns-settings__divider-mt-20"
  />
  <TabSubtitle
    text="Global settings"
  />
  <TabContainer>
    <Switch
      field="isMediaCollection"
      isChecked={false}
      switchHandler={[Function]}
      text="Enable media collection?"
    />
  </TabContainer>
  <div
    className="campaigns-settings__divider-mt-20 campaigns-settings__divider-mb-20 position-relative"
  >
    <TabSubtitle
      text="Accounts settings"
    />
    <AccountEditButton />
  </div>
  <AccountsSettingTable
    accountsSettingsItems={
      Array [
        Object {
          "id": "1",
          "mentionedFeed": false,
          "platform": "Instagram",
          "status": "Verified",
          "taggedFeed": true,
          "url": "test.png",
          "userFeed": true,
          "userName": "Instagram",
        },
        Object {
          "id": "2",
          "mentionedFeed": false,
          "platform": "Twitter",
          "status": "Rejected",
          "taggedFeed": false,
          "url": "test.png",
          "userFeed": false,
          "userName": "Blackmirror",
        },
      ]
    }
    stateHandler={[Function]}
  />
</Fragment>
`;

exports[`<Tab2 /> matches the snapshot 2`] = `
<Fragment>
  <TabTitle
    text="Media collection"
  />
  <div
    className="campaigns-settings__divider-mt-20"
  />
  <TabSubtitle
    text="Global settings"
  />
  <TabContainer>
    <Switch
      field="isMediaCollection"
      isChecked={true}
      switchHandler={[Function]}
      text="Enable media collection?"
    />
    <hr
      className="campaigns-settings__border-separator"
    />
    <CreatableSelectInput
      field="mediaCollectionList"
      itemsList={
        Array [
          "qwe",
          "rty",
        ]
      }
      limit={3}
      stateHandler={[Function]}
      text="Collect media with hashtags (max. 3, comma-separated)"
      tooltipText="test message"
    />
    <div
      className="campaigns-settings__divider-mt-32"
    />
    <Switch
      field="isPhotosModerationQueue"
      isChecked={true}
      switchHandler={[Function]}
      text="Hold incoming photos moderation queue?"
    />
  </TabContainer>
  <div
    className="campaigns-settings__divider-mt-20 campaigns-settings__divider-mb-20 position-relative"
  >
    <TabSubtitle
      text="Accounts settings"
    />
    <AccountEditButton />
  </div>
  <AccountsSettingTable
    accountsSettingsItems={
      Array [
        Object {
          "id": "1",
          "mentionedFeed": false,
          "platform": "Instagram",
          "status": "Verified",
          "taggedFeed": true,
          "url": "test.png",
          "userFeed": true,
          "userName": "Instagram",
        },
        Object {
          "id": "2",
          "mentionedFeed": false,
          "platform": "Twitter",
          "status": "Rejected",
          "taggedFeed": false,
          "url": "test.png",
          "userFeed": false,
          "userName": "Blackmirror",
        },
      ]
    }
    stateHandler={[Function]}
  />
</Fragment>
`;
